src/components/Map.tsx

```diff
@@ -72,27 +66,38 @@ const Map = ({
 
   const centerUserLocation = () => {
     camera.current?.setCamera({
-      centerCoordinate: currentLocation
-        ? [currentLocation.longitude, currentLocation.latitude]
+      centerCoordinate: userCoords
+        ? [userCoords.longitude, userCoords.latitude]
         : [-98.35, 15],
-      zoomLevel: currentLocation ? 16 : 2,
+      zoomLevel: userCoords ? 16 : 2,
       animationDuration: 500,
       heading: 0,
     })
   }
+  const flyTo = useCallback(
+    async (lat?: number, lng?: number, duration?: number) => {
+      if (!lat || !lng) return
+
+      await sleep(300) // this fixes a timing bug on Android where zoom level doesn't get set when flyTo is called to soon
```

Comment:
Where is the call to change the zoom level? This happens when you have animations and the next camera call happens before the current one finishes its animation. The better thing to do is to do both the zoom level and the flyTo in the same call to the camera so that you wont run into this problem. 

EG use `camera.current?.setCamera` like:
```
    camera.current?.setCamera({
      centerCoordinate: currentLocation
        ? [currentLocation.longitude, currentLocation.latitude]
        : [-98.35, 15],
      zoomLevel: currentLocation ? 16 : 2,
      animationDuration: 500,
      heading: 0,
    })
```

Reply:
I went ahead with this. Works well 👍